builders/ethereum/dev-env/thirdweb.md

```diff
@@ -110,184 +112,180 @@ thirdweb offers SDKs for a range of programming languages, such as React, React
 
-```bash
-npx thirdweb install
-```
+![thirdweb run locally](/images/builders/ethereum/dev-env/thirdweb/thirdweb-4.webp)
+
+### Create Client ID {: #create-client-id }
 
-### Initialize SDK On Moonbeam {: #initialize-sdk-on-moonbeam }
+If you generated your thirdweb app with Vite, you'll have a `client.ts` file that looks like the below. As long you've created a `.env` file with your thirdweb API Key (Client ID) defined in `VITE_TEMPLATE_CLIENT_ID`, you can leave the `client.ts` as is and proceed to the next section.
 
-Wrap your application in the `ThirdwebProvider` component and change the `activeChain` to Moonbeam.
+```typescript title="client.ts"
+import { createThirdwebClient } from 'thirdweb';
 
-```javascript
-import { ThirdwebProvider } from '@thirdweb-dev/react';
-import { Moonbeam } from '@thirdweb-dev/chains';
+// Replace this with your client ID string
+// refer to https://portal.thirdweb.com/typescript/v5/client on how to get a client ID
+const clientId = import.meta.env.VITE_TEMPLATE_CLIENT_ID;
 
-const App = () => {
-  return (
-    <ThirdwebProvider activeChain={Moonbeam}>
-      <YourApp />
-    </ThirdwebProvider>
-  );
-};
+export const client = createThirdwebClient({
+  clientId: clientId,
+});
 ```
 
+You can delete the placeholder code in `App.tsx` because we'll be replacing it as part of the tutorial. 
+
+### Configure Chain {: #configure-chain }
+
+thirdweb offers a small number of chains from `@thirdweb/chains` and does not include Moonbeam networks in that list, so you'll need to specify the network details including chain ID and RPC URL. You can create a custom chain with [`defineChain`](https://portal.thirdweb.com/references/typescript/v5/defineChain){target=\_blank} as follows:
+
+=== "Moonbeam"
+
+    ```typescript
+    import { defineChain } from 'thirdweb';
+    const moonbase = defineChain({
+    id: {{ networks.moonbeam.chain_id }},
+    rpc: '{{ networks.moonbeam.public_rpc_url }}',
+    })
+    ```
+
+=== "Moonriver"
+
+    ```typescript
+    import { defineChain } from 'thirdweb';
+    const moonbase = defineChain({
+    id: {{ networks.moonriver.chain_id }},
+    rpc: '{{ networks.moonriver.public_rpc_url }}',
+    })
+    ```
+
+=== "Moonbase Alpha"
+
+    ```typescript
+    import { defineChain } from 'thirdweb';
+    const moonbase = defineChain({
+    id: {{ networks.moonbase.chain_id }},
+    rpc: '{{ networks.moonbase.rpc_url }}',
+    })
+    ```
+
+
 ### Get Contract {: #get-contract }
 
-To connect to your contract, use the SDK’s [`getContract`](https://portal.thirdweb.com/typescript/sdk.thirdwebsdk.getcontract){target=\_blank} method.
+To connect to your contract, use the SDK’s [`getContract`](https://portal.thirdweb.com/references/typescript/v5/getContract){target=\_blank} method.  As an example, you can use Thirdweb to fetch data from an [Incrementer contract on Moonbase Alpha](https://moonbase.moonscan.io/address/0xa72f549a1a12b9b49f30a7f3aeb1f4e96389c5d8){target=\_blank}.
```

Comment:
I found the snippet that comes after this confusing. You mentioned deleting placeholder code, so I removed all of the jsx and stuck a "Hello world" div in there as a placeholder. My pain points from there:

- The snippet uses useState without showing useState in the imports, which causes an error. 
- It is unclear to me whether or not I was supposed to delete of all the starter code in App.tsx (rather than just the jsx) before adding your snippet. If I am supposed to replace everything with your code snippet, you are missing a closing bracket for the App(), which causes an error. 
- Getting a type error for `abi: ABI" in the const myContract toward the bottom of the snippet